nicegui/outbox.py

```diff
@@ -23,35 +23,46 @@ def __init__(self, client: Client) -> None:
         self.updates: Dict[ElementId, Optional[Element]] = {}
         self.messages: Deque[Message] = deque()
         self._should_stop = False
+        self._enqueue_event = None
         if core.app.is_started:
             background_tasks.create(self.loop(), name=f'outbox loop {client.id}')
         else:
             core.app.on_startup(self.loop)
 
+    def _set_enqueue_event(self):
+        """Set the enqueue event while accounting for lazy initialization."""
+        if self._enqueue_event:
+            self._enqueue_event.set()
+
     def enqueue_update(self, element: Element) -> None:
         """Enqueue an update for the given element."""
         self.updates[element.id] = element
+        self._set_enqueue_event()
 
     def enqueue_delete(self, element: Element) -> None:
         """Enqueue a deletion for the given element."""
         self.updates[element.id] = None
+        self._set_enqueue_event()
 
     def enqueue_message(self, message_type: MessageType, data: Any, target_id: ClientId) -> None:
         """Enqueue a message for the given client."""
         self.messages.append((target_id, message_type, data))
+        self._set_enqueue_event()
 
     async def loop(self) -> None:
         """Send updates and messages to all clients in an endless loop."""
+        self._enqueue_event = asyncio.Event()
+        self._enqueue_event.set()
+
         while not self._should_stop:
             try:
-                await asyncio.sleep(0.01)
-
-                if not self.updates and not self.messages:
-                    continue
-
+                await self._enqueue_event.wait()
+                await asyncio.sleep(0.005)
```

Comment:
Why do we need to sleep after the event has been awaited?

Reply:
With the current implementation, there is some batching of queue processing due to the sleep interval. I wasn't sure if this was an intentional part of the design, so I first left it out in order to minimize latency in the system. During automated testing, this led to an increase in the number of random errors. I surmised that this increase in errors was due to the user interface being less responsive without the batching behavior, leading to more missed input events and therefore random errors. When adding it back, I chose 5ms as the batch duration to approximate the average delay in processing with the current implementation. This prevented any increase in errors. If the sleep is to be left out, some adjustment of screen.wait() calls in the tests may be necessary.

Reply:
I had some more thoughts on this. I chose to include the batching because of the issues I was seeing in testing and out of an abundance of caution. Despite this, I'm not completely convinced of its necessity. 

Firstly, my test setup is relatively low-performance. It involves multiple layers of virtualization on a six-year-old CPU. This may have made the tests more sensitive to small changes in timing. It's quite possible you'd see no issues whatsoever on your typical test systems. It also occurred to me that the increased test failures might not be an indication of overall lower performance but instead the result of the test timings being tuned to the current implementation.

Secondly, in my ad hoc testing of my own GUIs, I didn't notice any difference in performance with or without the batching. This leads me to believe that whatever small difference there is, it might be negligible from the user standpoint.

I'm on the fence about leaving it in since, now that you're aware of the potential issue, it can trivially be added later if problems arise in your testing. Let me know what you think.

Reply:
Is it safe to `asyncio.sleep(0.005)` here given that we're removing `asyncio.sleep(0.01)` or replacing it with `asyncio.sleep(0.016)` to avoid issues with sleep interval being less than Windows timer resolution?

Reply:
I think it was safe to remove this `sleep` in https://github.com/zauberzeug/nicegui/pull/2867/commits/a9d0d52bc01723bee0ad7d53d00061ab3d2114ca. But let me know if it works on your machine.

Reply:
Are you seeing increased test failures? I see you changed some tests.

Reply:
@me21 
> Is it safe to `asyncio.sleep(0.005)` here given that we're removing `asyncio.sleep(0.01)` or replacing it with `asyncio.sleep(0.016)` to avoid issues with sleep interval being less than Windows timer resolution?

It would be safe here because the `asyncio.Event` is what controls execution of the loop, so there's no chance of it running out of control, even with 0 sleep. It was only included as a possible mitigation against an increase in test failures.